src/addons/transitions/ReactTransitionChildMapping.js

```diff
@@ -25,6 +25,9 @@ var ReactTransitionChildMapping = {
     if (!children) {
       return children;
     }
+    // TODO: `flattenChildren` now takes an extra `selfDebugID` argument
```

Comment:
Also, seems like the only other place we use `flattenChildren` is in `ReactMultiChild` and there's already a test for that in `ReactMultiChild-test`. Interestingly, when `ReactChildReconciler` finds duplicated keys, the [error message](https://github.com/facebook/react/blob/master/src/renderers/shared/stack/reconciler/ReactChildReconciler.js#L28) talks about "flattenChildren".